src/Microsoft.Framework.Configuration.File/Helper/ConfigurationHelper.cs

```diff
@@ -10,18 +10,13 @@ public static class ConfigurationHelper
     {
         public static string ResolveConfigurationFilePath(IConfigurationBuilder configuration, string path)
         {
-            if (!Path.IsPathRooted(path))
+            if (configuration.Properties.ContainsKey("BasePath"))
             {
-                if (configuration.BasePath == null)
-                {
-                    throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        path,
-                        typeof(IConfigurationBuilder).Name,
-                        nameof(configuration.BasePath)));
-                }
-                else
+                string basePath = configuration.Properties["BasePath"].ToString();
```

Comment:
I think we should just do a `(string)` cast. If it fails then the wrong data got into the property somehow.